google-collab-version/PytorchDigitalPathology/segmentation_epistroma_unet/.ipynb_checkpoints/make_output_unet_cmd-checkpoint.ipynb

```diff
@@ -0,0 +1,650 @@
+    "id": "GcFBci8NAsXT"
+   },
+   "outputs": [],
+   "source": [
+    "checkpoint = torch.load(args.model, map_location=lambda storage, loc: storage) \n",
+    "#load checkpoint to CPU and then put to device https://discuss.pytorch.org/t/saving-and-loading-torch-models-on-2-machines-with-different-number-of-gpu-devices/6666"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "t2tqNu5GerD9",
+    "outputId": "fd3760c7-a0f6-4a84-9dcf-38d2573dcc8c"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "total params: \t122466\n"
+     ]
+    }
+   ],
+   "source": [
+    "model = UNet(n_classes=checkpoint[\"n_classes\"], in_channels=checkpoint[\"in_channels\"],\n",
+    "             padding=checkpoint[\"padding\"], depth=checkpoint[\"depth\"], wf=checkpoint[\"wf\"],\n",
+    "             up_mode=checkpoint[\"up_mode\"], batch_norm=checkpoint[\"batch_norm\"]).to(device)\n",
+    "model.load_state_dict(checkpoint[\"model_dict\"])\n",
+    "model.eval()\n",
+    "\n",
+    "print(f\"total params: \\t{sum([np.prod(p.size()) for p in model.parameters()])}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "id": "_Sp1C2x9fI2X"
+   },
+   "outputs": [],
+   "source": [
+    "# ----- get file list\n",
+    "\n",
+    "if not os.path.exists(OUTPUT_DIR):\n",
+    "    os.makedirs(OUTPUT_DIR)\n",
+    "\n",
+    "files = []\n",
+    "basepath = args.basepath  #\n",
+    "basepath = basepath + os.sep if len(\n",
+    "    basepath) > 0 else \"\"  # if the user supplied a different basepath, make sure it ends with an os.sep\n",
+    "\n",
+    "if len(args.input_pattern) > 1:  # bash has sent us a list of files\n",
+    "    files = args.input_pattern\n",
+    "elif args.input_pattern[0].endswith(\"tsv\"):  # user sent us an input file\n",
+    "    # load first column here and store into files\n",
+    "    with open(args.input_pattern[0], 'r') as f:\n",
+    "        for line in f:\n",
+    "            if line[0] == \"#\":\n",
+    "                continue\n",
+    "            files.append(basepath + line.strip().split(\"\\t\")[0])\n",
+    "else:  # user sent us a wildcard, need to use glob to find files\n",
+    "    if len(args.basepath)>=1:\n",
+    "      files = glob.glob(args.basepath +\"/\"+ args.input_pattern[0])\n",
+    "    else:\n",
+    "      files = glob.glob(args.input_pattern[0])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "id": "65Y7Qgfa6W79"
+   },
+   "outputs": [],
+   "source": [
+    "#Copying the supporting PS_scikitlearn.py to the colab notebook runtime. This has the extract_patches function which is now a private method.\n",
+    "!cp '/gdrive/My Drive/PytorchDigitalPathology/segmentation_epistroma_unet/PS_scikitlearn.py' /content/PS_scikitlearn.py\n",
```

Comment:
Here "classification_lymphoma_densenet" is replaced to "segmentation_epistroma_unet"